riptide-opentelemetry/src/main/java/org/zalando/riptide/opentelemetry/span/ErrorSpanDecorator.java

```diff
@@ -0,0 +1,35 @@
+package org.zalando.riptide.opentelemetry.span;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import org.springframework.http.client.ClientHttpResponse;
+import org.zalando.riptide.RequestArguments;
+
+import java.io.IOException;
+import java.util.function.Predicate;
+
+import static org.zalando.fauxpas.FauxPas.throwingPredicate;
+
+public class ErrorSpanDecorator implements SpanDecorator {
+    private final Predicate<ClientHttpResponse> predicate;
```

Comment:
The way this is setup I don't think you really need the predicate. It is set by the constructor, with no way for the user to override it. In the OpenTracing plugin users can override the predicate for the [`ErrorSpanDecorator`](https://github.com/zalando/riptide/blob/main/riptide-opentracing/src/main/java/org/zalando/riptide/opentracing/span/ErrorSpanDecorator.java) (it is annotated with `@AllArgsConstructor`). Either we go with something along the same lines, or we make this one simpler by removing the predicate. I would favor consistency across Plugins, and follow the same approach.